File: pset6/caesar/caesar.py
from sys import argv
from cs50 import get_string


def main():
    validate_argv()
    text = prompt_text()
    key = int(argv[1])
    print("ciphertext:", encrypt_caesar(text, key))


# Validates command correctness
def validate_argv():
    if len(argv) != 2:
        print("Usage: python caesar.py k")
        exit(1)


# Prompt user for plain text
def prompt_text():
    return get_string("plaintext: ")


def encrypt_caesar(text, key):
    encrypted = ""
    for c in text:
        # Modify only letters
        if c.isalpha():
            # Preserve case
            shift = ord('A') if c.isupper() else ord('a')
            # Calculate new position of a character
            new_position = (ord(c) - shift + key) % 26
            encrypted += chr(new_position + shift)
        else:
            encrypted += c
    return encrypted


if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

string encipher_caesar(string, int);

int main(int argc, string argv[])
{
    // Check arguments
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    int key = atoi(argv[1]);

    // Take user input text
    string text = get_string("plaintext: ");

    // Output the ciphered text
    printf("ciphertext: %s\n", encipher_caesar(text, key));

    return 0;
}

string encipher_caesar(string text, int key)
{
    // For shifting to/from zero starting position
    int shift;
    for (int i = 0, n = strlen(text); i < n; i ++)
    {
        if (isalpha(text[i]))
        {
            shift = isupper(text[i]) ? 'A' : 'a';
            text[i] = (text[i] - shift + key) % 26 + shift;
        }
    }
    return text;
}